Sources/_StringProcessing/ConsumerInterface.swift

```diff
@@ -123,7 +123,7 @@ extension AST.Atom {
 extension AST.CustomCharacterClass.Member {
   func generateConsumer(
     _ opts: MatchingOptions
-  ) throws -> Program<String>.ConsumeFunction {
+  ) throws -> Program<String>.ConsumeFunction? {
```

Comment:
I'd rather strip trivia first and error (i.e. throw unreachable) if we see it. 

How about a `var strippingTrivia: Self` on `CustomCharacterClass`?

Reply:
How does this look @milseman?